packages/firestore/src/core/sync_engine.ts

```diff
@@ -138,7 +138,11 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
     [uidKey: string]: SortedMap<BatchId, Deferred<void>>;
   };
   private limboTargetIdGenerator = TargetIdGenerator.forSyncEngine();
-  private isPrimary = false;
+
+  // The primary state is set to `true` or `false` immediately after Firestore
+  // startup. In the interim, a client should only be considered primary if `
+  // isPrimary` is true.
```

Comment:
Line break fixed.

I opted to not change the current behavior for `isPrimary`. Calling `disableNetwork` in the constructor causes NPEs, calling in start() (instead of enable network) causes a bunch of test failures since `applyPrimaryState` might have been called beforehand, and passing the primary state into the constructor/start() is not an easy task since it needs to be read from IndexedDb. Sorry :/